src/python/WMCore/WMBS/MySQL/Workflow/GetDeletableWorkflows.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+"""
+_GetDeletableWorkflows_
+
+MySQL implementation of Workflows.GetFinishedWorkflows
+
+Created on Aug 29, 2012
+
+@author: dballest
+"""
+
+from WMCore.Database.DBFormatter import DBFormatter
+
+class GetDeletableWorkflows(DBFormatter):
+    """
+    A workflow is finished when all subscriptions associated to it are finished
+    and all the other workflows using the same spec are finished as well,
+    this DAO searches such workflows. It assumes that a spec is not shared by
+    two workflows of the same name.
+    """
+    deletableWFSql = """SELECT wmbs_workflow.name, wmbs_workflow.spec,
+                               wmbs_workflow.id AS workflow_id, 
+                               wmbs_subscription.id AS sub_id
+                                  FROM wmbs_workflow
+                                      LEFT OUTER JOIN wmbs_subscription ON
+                                         wmbs_subscription.workflow = wmbs_workflow.id AND
+                                         wmbs_subscription.finished = 0
+                                      INNER JOIN wmbs_fileset ON
+                                         wmbs_fileset.id = wmbs_subscription.fileset
+                                      LEFT OUTER JOIN wmbs_fileset_files ON
+                                         wmbs_fileset_files.fileset = wmbs_fileset.id
+                                      LEFT OUTER JOIN wmbs_file_parent ON
+                                         wmbs_file_parent.parent = wmbs_fileset_files.fileid
+                                      LEFT OUTER JOIN wmbs_fileset_files child_fileset ON
+                                         child_fileset.fileid = wmbs_file_parent.child
+                                      LEFT OUTER JOIN wmbs_subscription child_subscription ON
+                                         child_subscription.fileset = child_fileset.fileset AND
+                                         child_subscription.finished = 0
+                                      LEFT OUTER JOIN wmbs_workflow child_workflow ON
+                                         child_subscription.workflow = child_workflow.id AND
+                                         child_workflow.name != wmbs_subscription.name
+                                  GROUP BY wmbs_workflow.name
+                                  HAVING COUNT(wmbs_subscription.workflow) = 0  AND COUNT(child_workflow.name) = 0
```

Comment:
This "COUNT(child_workflow.name) = 0" only evaluates to true when the workflow has been completely deleted from WMBS, which will never be the case here (I guess killWorkflow() does that job), right?